scripts/e2etests/VSUtils.ps1

```diff
@@ -169,6 +169,45 @@ function UninstallVSIX
     return $true
 }
 
+function DowngradeVSIX
+{
+    param(
+        [Parameter(Mandatory=$true)]
+        [string]$vsixID,
+        [Parameter(Mandatory=$true)]
+        [ValidateSet("15.0")]
+        [string]$VSVersion,
+        [Parameter(Mandatory=$true)]
+        [int]$VSIXInstallerWaitTimeInSecs
+    )
+
+    $VSIXInstallerPath = GetVSIXInstallerPath $VSVersion
+
+    Write-Host 'Downgrading VSIX...'
+    $p = start-process "$VSIXInstallerPath" -Wait -PassThru -NoNewWindow -ArgumentList "/q /a /d:$vsixID"
+
+    if ($p.ExitCode -ne 0)
+    {
+        if($p.ExitCode -eq 1002)
```

Comment:
Have you encountered this exception while downgrading?
I had added this check in the `UnInstallVSIX `function before we started building v3 vsix and then it was possible to be not have the vsix installed. I am not sure if you can get this exception while downgrading.